src/Bridge/Symfony/Routing/IriConverter.php

```diff
@@ -44,24 +44,23 @@
     private $routeNameResolver;
     private $router;
     private $identifiersExtractor;
-    private $identifierDenormalizer;
+    private $operationDataProvider;
 
-    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ItemDataProviderInterface $itemDataProvider, RouteNameResolverInterface $routeNameResolver, RouterInterface $router, PropertyAccessorInterface $propertyAccessor = null, IdentifiersExtractorInterface $identifiersExtractor = null, ChainIdentifierDenormalizer $identifierDenormalizer = null)
+    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ItemDataProviderInterface $itemDataProvider, RouteNameResolverInterface $routeNameResolver, RouterInterface $router, PropertyAccessorInterface $propertyAccessor = null, IdentifiersExtractorInterface $identifiersExtractor = null, OperationDataProvider $operationDataProvider = null)
     {
         $this->itemDataProvider = $itemDataProvider;
         $this->routeNameResolver = $routeNameResolver;
         $this->router = $router;
-        $this->identifierDenormalizer = $identifierDenormalizer;
+        $this->identifiersExtractor = $identifiersExtractor;
+        $this->operationDataProvider = $operationDataProvider;
 
         if (null === $identifiersExtractor) {
             @trigger_error('Not injecting ItemIdentifiersExtractor is deprecated since API Platform 2.1 and will not be possible anymore in API Platform 3', E_USER_DEPRECATED);
             $this->identifiersExtractor = new IdentifiersExtractor($propertyNameCollectionFactory, $propertyMetadataFactory, $propertyAccessor ?? PropertyAccess::createPropertyAccessor());
-        } else {
-            $this->identifiersExtractor = $identifiersExtractor;
         }
 
-        if (null === $identifierDenormalizer) {
-            @trigger_error(sprintf('Not injecting "%s" is deprecated since API Platform 2.2 and will not be possible anymore in API Platform 3.', ChainIdentifierDenormalizer::class), E_USER_DEPRECATED);
+        if (null === $operationDataProvider) {
```

Comment:
It will make very difficult to use API Platform without a framework or with another framework than Symfony. For instance, if I want to use API Platform's classes without a framework, and I don't bother to support subresources (I've a project like that), I'm stuck.